src/providers/dataProvider/index.ts

```diff
@@ -262,9 +262,9 @@ export const dataProvider = (apiUrl: string): DataProvider => ({
     // remove the id property
     const { id, ...editData } = params.data
 
-    return await axios.put(url, { fields: editData }).then((response) => {
+    return await axios.put(url, { fields: editData }).then(() => {
       return {
-        data: { id: response.data.data.lastInsertRowid, ...params.data }
+        data: { id: params.id, ...params.data }
```

Comment:
haven't we extracted `id` as a `const` in line 263 @TahaKhanAbdalli ?

Reply:
No, the `id` is `undefined` as we are only sending `reportPrintedAt` prop to update function.